docs/getting-started-provider-dev.md

```diff
@@ -0,0 +1,221 @@
+
+```
+sed -i '/^ExecStart=\/usr\/bin\/dockerd$/ s/$/ --exec-opt native.cgroupdriver=systemd/' \
+       /usr/lib/systemd/system/docker.service
+```
+
+You may want to configure your environment to allow you to control Docker without sudo:
+
+```
+user="$(id -un 2\>/dev/null || true)"
+sudo usermod -aG docker centos
+```
+
+Regardless of how you install, enable start the service:
+
+```
+sudo systemctl daemon-reload
+sudo systemctl enable docker
+sudo systemctl start docker
+```
+
+### Development tools
+
+You're going to need a few basic development tools and applications to get, build, and run
+the source code. With your package manager you can install `git`, `gcc`, `glide`, and `etcd`.
+
+```
+sudo yum install -y -q git gcc glide etcd
+```
+
+You will also need a recent version of Go and set your environment variables.
+
+```
+GO_VERSION=1.10
+GO_ARCH=linux-amd64
+curl -o go.tgz https://dl.google.com/go/go$(GO_VERSION).$(GO_ARCH).tar.gz
+sudo tar -C /usr/local/ -xvzf go.tgz
+export GOROOT=/usr/local/go
+export GOPATH=$HOME/go
+```
+
+Finally, set up your Git identity and GitHub integrations.
+
+More comprehensive setup instructions are available in the 
+[Development Guide](https://github.com/kubernetes/community/blob/master/contributors/devel/development.md#building-kubernetes-on-a-local-osshell-environment)
+in the Kubernetes repository. When in doubt, check there for additional setup
+and versioning information.
+
+## Development
+
+### Getting and Building Cloud Provider OpenStack
+
+Following the [GitHub Workflow](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md)
+guidlines for Kubernetes development, set up your environment and get the latest development repository. Begin
+by forking both the Kubernetes and Cloud-Provider-OpenStack projects into your GitHub into your local
+workspace (or bringing your current fork up to date with the current state of both repositories).
+
+Set up some environment variables to help download the repositories
+
+```
+export GOPATH=$HOME/go
+export GOROOT=/usr/local/go
+export PATH=$PATH:$GOROOT/bin
+export user={your github profile name}
+export working_dir=$GOPATH/src/k8s.io
+```
+
+With your environment variables set up, clone the forks into your go environment.
+
+```
+mkdir -p $working_dir
+cd $working_dir
+git clone https://github.com/{user}/cloud-provider-openstack
+cd cloud-provider-openstack
+```
+
+If you want to build the provider:
+
+```
+make
+```
+
+If you want to run unit tests:
+
+```
+make test
+```
+
+### Getting and Building Kubernetes
+
+To get and build Kubernetes
+
+```
+cd $working_dir
+git clone https://github.com/{user}/kubernetes
+cd kubernetes
+make
+```
+
+### Running the Cloud Provider in MiniKube
+
+To run the OpenStack provider, integrated with your cloud, be sure to have sourced the
+environment variables. You will also need to create an `/etc/kubernetes/cloud-config` file
+with the minimum options:
```

Comment:
we can throw the url for the configuration options here too https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/#cloudconf

Reply:
That document is going away, and we should be hosting it in this repository.